lib/glueby/contract/token_tx_builder.rb

```diff
@@ -0,0 +1,201 @@
+# frozen_string_literal: true
+
+module Glueby
+  module Contract
+    module TokenTxBuilder
+      def receive_address(wallet:)
+        wallet.receive_address
+      end
+
+      # Create new public key, and new transaction that sends TPC to it
+      def create_funding_tx(wallet:, amount:, script: nil, fee_provider: FixedFeeProvider.new)
+        tx = Tapyrus::Tx.new
+        fee = fee_provider.fee(tx)
+
+        utxos = wallet.internal_wallet.list_unspent
+        sum, outputs = collect_uncolored_outputs(utxos, fee + amount)
+        fill_input(tx, outputs)
+
+        receiver_script = script ? script : Tapyrus::Script.parse_from_addr(wallet.internal_wallet.receive_address)
+        tx.outputs << Tapyrus::TxOut.new(value: amount, script_pubkey: receiver_script)
+
+        fill_change_tpc(tx, wallet, sum - fee - amount)
+        wallet.internal_wallet.sign_tx(tx)
+      end
+
+      def create_issue_tx_for_reissuable_token(funding_tx:, issuer:, amount:, fee_provider: FixedFeeProvider.new)
+        tx = Tapyrus::Tx.new
+        fee = fee_provider.fee(tx)
+
+        out_point = Tapyrus::OutPoint.from_txid(funding_tx.txid, 0)
+        tx.inputs << Tapyrus::TxIn.new(out_point: out_point)
+        output = funding_tx.outputs.first
+
+        receiver_script = Tapyrus::Script.parse_from_payload(output.script_pubkey.to_payload)
+        color_id = Tapyrus::Color::ColorIdentifier.reissuable(receiver_script)
+        receiver_colored_script = receiver_script.add_color(color_id)
+        tx.outputs << Tapyrus::TxOut.new(value: amount, script_pubkey: receiver_colored_script)
+
+        fill_change_tpc(tx, issuer, output.value - fee)
+        prev_txs = [{
+          txid: funding_tx.txid,
+          vout: 0,
+          scriptPubKey: output.script_pubkey.to_hex,
+          amount: output.value
+        }]
+        issuer.internal_wallet.sign_tx(tx, prev_txs)
+      end
+
+      def create_issue_tx_for_non_reissuable_token(issuer:, amount:, fee_provider: FixedFeeProvider.new)
+        tx = Tapyrus::Tx.new
+        fee = fee_provider.fee(tx)
+
+        utxos = issuer.internal_wallet.list_unspent
+        sum, outputs = collect_uncolored_outputs(utxos, fee)
+        fill_input(tx, outputs)
+
+        out_point = tx.inputs.first.out_point
+        color_id = Tapyrus::Color::ColorIdentifier.non_reissuable(out_point)
+
+        receiver_script = Tapyrus::Script.parse_from_addr(issuer.internal_wallet.receive_address)
+        receiver_colored_script = receiver_script.add_color(color_id)
+        tx.outputs << Tapyrus::TxOut.new(value: amount, script_pubkey: receiver_colored_script)
+
+        fill_change_tpc(tx, issuer, sum - fee)
+        issuer.internal_wallet.sign_tx(tx)
+      end
+
+      def create_issue_tx_for_nft_token(issuer:, fee_provider: FixedFeeProvider.new)
+        tx = Tapyrus::Tx.new
```

Comment:
`create_issue_tx_for_nft_token` is almost same as  `create_issue_tx_for_non_reissuable_token`, so code can be shared.

Reply:
fix at 500ac64afee43573fad32930169c45619ca664c3